test/e2e/framework/service/jig.go

```diff
@@ -764,17 +764,31 @@ func testReachabilityOverClusterIP(clusterIP string, sp v1.ServicePort, execPod
 	return nil
 }
 
-func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v1.Pod) error {
+func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v1.Pod, clusterIP string) error {
 	internalAddrs := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
 	externalAddrs := e2enode.CollectAddresses(nodes, v1.NodeExternalIP)
+	reIPV4, err := regexp.Compile(e2enetwork.RegexIPv4)
+	if err != nil {
+		return fmt.Errorf("unable to compile regular exp: %v", err)
+	}
+	isClusterIPV4 := reIPV4.MatchString(clusterIP)
+
 	for _, internalAddr := range internalAddrs {
 		// If the node's internal address points to localhost, then we are not
 		// able to test the service reachability via that address
 		if isInvalidOrLocalhostAddress(internalAddr) {
 			framework.Logf("skipping testEndpointReachability() for internal adddress %s", internalAddr)
 			continue
 		}
-		err := testEndpointReachability(internalAddr, sp.NodePort, sp.Protocol, pod)
+		isNodeInternalIPV4 := reIPV4.MatchString(internalAddr)
+		// Check service reachability on the node internalIP which is same family
+		// as clusterIP
+		if isClusterIPV4 != isNodeInternalIPV4 {
+			framework.Logf("skipping testEndpointReachability() for internal adddress %s as does not match cluster ip (%s) family", internalAddr, clusterIP)
```

Comment:
`as does not match` -> `as it does not match`

Reply:
`ip` -> `IP`

Reply:
updated!